### PR TITLE
[IOTDB-3178] Refactor integration test for the new cluster

### PR DESCRIPTION
**Pull request by irvine0109**
New feature:
- Adapt for the new cluster of MPP.
- Add the log record for every node in the cluster integration test.
- Parallel to do cluster integration test.

TODO:
- [x] Add the whitelist for IT files in this module.
- [x] The log record files can be downloaded.
- [x] To integrate with GitHub action.
- [x] Finish the README. 
- [ ] Finish the user document.

Use command below to do test:
`mvn spotless:apply clean verify -Dsession.test.skip=true -Diotdb.test.skip=true -Dcluster.test.skip=true -Dtsfile.test.skip=true -Dcommons.test.skip=true -Dconfignode.test.skip=true -pl integration-test -am -PClusterIT`

### CODE REVIEW

File: integration-new/pom.xml [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+            <artifactId>iotdb-server</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.iotdb</groupId>
+            <artifactId>iotdb-server</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
```
Comment (qiaojialin) [inline integration-new/pom.xml:43]: Duplicated?

**Review by qiaojialin — COMMENTED** [review]
Hi, the picture could be put into https://github.com/apache/iotdb-bin-resources

File: integration-new/pom.xml (hunk 1 above)
Comment (irvine0109) [reply integration-new/pom.xml:43]: This file is a copy of the integration module. And line 37-43 is introduced by [this commit](https://github.com/apache/iotdb/commit/3dca44a4eab975aa8e0ddeb31067761f748078ba). I don't understand this part's meaning, and I will figure this out with the author.

**Comment by irvine0109** [comment]
> Hi, the picture could be put into https://github.com/apache/iotdb-bin-resources. 

Thanks, I will put the pictures into the new place.   
These pictures are all out of date. The README and user document will be updated later.

**Comment by qiaojialin** [comment]
@irvine0109  Hi, Haonan should already fixed the pictures~

File: .github/workflows/cluster-new.yml [hunk 2]
```diff
@@ -49,4 +51,12 @@ jobs:
...
         # we can skip influxdb-protocol because it has been tested separately in influxdb-protocol.yml
-        run: mvn -B clean verify -Dsession.test.skip=true -Diotdb.test.skip=true -Dcluster.test.skip=true -Dtsfile.test.skip=true -pl integration -am -PCluster
+        run: mvn -B clean verify -Dsession.test.skip=true -Diotdb.test.skip=true -Dcluster.test.skip=true -Dtsfile.test.skip=true -Dcommons.test.skip=true -Dconfignode.test.skip=true -pl integration-new -am -PNewCluster
+      - name: Upload Artifact
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: cluster-log
+          path: integration-new/target/cluster-logs
+          retention-days: 1
```
Comment (HTHou) [inline .github/workflows/cluster-new.yml:61]: This will upload the test log files?

File: confignode/pom.xml [hunk 3]
```diff
@@ -50,11 +50,6 @@
             <artifactId>iotdb-server</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
-            <groupId>org.apache.iotdb</groupId>
-            <artifactId>iotdb-procedure</artifactId>
-            <version>${project.version}</version>
-        </dependency>
```
Comment (HTHou) [inline confignode/pom.xml:57]: Why remove procedure here?

Comment (irvine0109) [reply confignode/pom.xml:57]: The procedure module is removed in [this commit](https://github.com/apache/iotdb/commit/fa06bbb43243fef53fa07b82dc848cd3684e09b4). IDE reported an error message about this part.

File: .github/workflows/cluster-new.yml (hunk 2 above)
Comment (irvine0109) [reply .github/workflows/cluster-new.yml:61]: Yes. This upload step only happens after the previous IT step fails.
You can download the zip file of log files at [this URL](https://github.com/irvine0109/iotdb/actions/runs/2333221271). The Artifacts part is at the bottom.

File: .github/workflows/cluster-new.yml [hunk 4]
```diff
@@ -1,14 +1,16 @@
-name: Cluster Test
+name: New Cluster IT
 
 on:
   push:
     branches:
       - test_cluster
+      - IOTDB-3178*
```
Comment (HTHou) [inline .github/workflows/cluster-new.yml:7]:
```suggestion
      - master
```

File: .github/workflows/cluster-new.yml [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
     paths-ignore:
       - 'docs/**'
   pull_request:
     branches:
       - test_cluster
+      #- master
```
[the remark this answers is not in the document]
Comment (irvine0109) [reply .github/workflows/cluster-new.yml]: done

**Review by qiaojialin — APPROVED** [review]

**Review by HTHou — APPROVED** [review]

**Pull request merged**